db/migrate/20151111132037_remove_latitude_and_longitude_from_firm.rb

```diff
@@ -0,0 +1,72 @@
+class RemoveLatitudeAndLongitudeFromFirm < ActiveRecord::Migration
+  class Firm < ActiveRecord::Base
+    has_many :offices, -> { order created_at: :asc }
+    scope :geocoded, -> { where.not(latitude: nil, longitude: nil) }
+
+    def main_office
+      offices.first
+    end
+  end
+
+  class Office < ActiveRecord::Base
+    belongs_to :firm
+
+    def geocoded?
+      latitude.present? && longitude.present?
+    end
+  end
+
+  def up
+    migrate_firm_coords_to_main_office
+
+    remove_column :firms, :latitude
+    remove_column :firms, :longitude
+
+    index_warning
+  end
+
+  def down
+    add_column :firms, :latitude, :float
+    add_column :firms, :longitude, :float
+
+    migrate_main_office_coords_to_firm
+
+    index_warning
+  end
+
+  private
+
+  def migrate_firm_coords_to_main_office
+    update_count = 0
+
+    Firm.geocoded.includes(:offices).find_each do |f|
+      next unless f.offices.present?
+
+      f.main_office.update!(latitude: f.latitude,
+                            longitude: f.longitude)
+
+      update_count += 1
+    end
+
+    say "Migrated #{update_count} coordinates to main offices"
```

Comment:
when you run this on the server the will say hello? have you managed to someone connect speakers to the server or pipe the audio to your local machine